tools/shared/src/main/scala/org/scalajs/core/tools/io/VirtualFiles.scala

```diff
@@ -118,6 +118,20 @@ trait WritableVirtualJSFile extends WritableVirtualTextFile with VirtualJSFile {
   def sourceMapWriter: Writer
 }
 
+/** An virtual file containing Scala.js IR.
+ *
+ *  This can be a VirtualScalaJSIRFile (with RelativeVirtualFile) or a
+ *  VirtualFileContainer.
+ *
+ *  The main difference to using individual files
+ *  (that are extracted beforehand) is that the fileset can be versioned at a
+ *  higher level: The container needs to change its version when any of the
+ *  files change. Therefore, the entire extraction process can be cached.
+ */
+trait IRContainer extends VirtualFile {
```

Comment:
To be consistent with the rest of the virtual file API, this should be called `ScalaJSIRContainer`. We always write `ScalaJSIR` rather than just `IR`.